src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -313,6 +314,43 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Adds a new telemetry sink to the configuration.
+        /// </summary>
+        /// <param name="sink">Sink to add.</param>
+        public void AddSink(TelemetrySink sink)
```

Comment:
why do you need a method for this? Shouldn't it be added via configuration the same way it works for telemetry initializers?